nntrainer/graph/network_graph.cpp

```diff
@@ -576,8 +576,22 @@ void NetworkGraph::inPlaceOptimize(const std::string &layer_type,
         continue;
 
       /** Share tensor with next layer */
-      l->net_input[0] = l->net_hidden[0];
-      prev_layer->net_hidden[loc] = l->net_hidden[0];
+      l->net_input[0] = l->net_hidden[0]; // I2 = O2
+      if (l->getType() == BatchNormalizationLayer::type) {
+        prev_layer->net_hidden[loc] = l->net_hidden[0]; // O1 = O2
+      } else if (l->getType() == ActivationLayer::type) {
+        // THIS IS NOT WORKING BECAUSE INITIALIZES MAKES NEW SHARED_PTR OF TENSOR
+        // THAN KEEPING THE SAME SHARED_PTR AND JUST CHANGING THE TENSOR IN IT
+        prev_layer->net_hidden[loc]->getGradientRef() =
+          l->net_hidden[0]->getVariableRef(); // O1.G = O2.V
```

Comment:
How about having a designated memory(for safety) or at least alias for readability?

Reply:
designated memory(for safety) -> you mean a designated variable?

Reply:
yup. sth like `Tensor inPlaceSharedSlot`

or `Tensor & in_place_shared_slot = l->net_hidden[0]->getVariableRef()` for readability

Reply:
updated.